test/FunctionalTests/Microsoft.OData.Client.Tests/Tracking/DataServiceContextQueryTests.cs

```diff
@@ -0,0 +1,274 @@
+            Assert.Equal(2, employees.Count());
+        }
+
+        [Fact]
+        public void UseWhereToFilterByEmployeeNumberKey_WithEnumAsKey_DoNotThrowException()
+        {
+            // Arrange
+            string expectedUri = $"{ServiceRoot}/Employees?$filter=EmpNumber eq 8";
+
+            string response = @"{
+    ""@odata.context"": ""http://localhost:8007/$metadata#Employees"",
+    ""value"": [
+        {
+            ""EmpNumber"": 8,
+            ""EmpType"": ""PartTime"",
+            ""OrgId"": 1,
+            ""Name"": ""Employee Two""
+        }
+    ]
+}";
+            SetupContextWithRequestPipeline(new DataServiceContext[] { _defaultContext }, response, "Employees");
+
+            // Act
+            IQueryable<Employee> query = _defaultContext.Employees.Where(e => e.EmpNumber == 8);
+            Employee employee = query.First();
+
+            // Assert
+            Assert.Equal(expectedUri, query.ToString());
+            Assert.Equal(EmployeeType.PartTime, employee.EmpType);
+            Assert.Equal("Employee Two", employee.Name);
+        }
+
+        [Fact]
+        public void UseWhereToFilterByEnumKey_WithEnumAsKey_DoNotThrowException()
+        {
+            // Arrange
+            string expectedUri = $"{ServiceRoot}/Employees?$filter=EmpType eq Microsoft.OData.Client.Tests.Tracking.EmployeeType'PartTime'";
+
+            string response = @"{
+    ""@odata.context"": ""http://localhost:8007/$metadata#Employees"",
+    ""value"": [
+        {
+            ""EmpNumber"": 8,
+            ""EmpType"": ""PartTime"",
+            ""OrgId"": 1,
+            ""Name"": ""Employee 45""
+        }
+    ]
+}";
+            SetupContextWithRequestPipeline(new DataServiceContext[] { _defaultContext }, response, "Employees");
+
+            // Act
+            var query = _defaultContext.Employees.Where(e => e.EmpType == EmployeeType.PartTime);
+            Employee employee = query.First();
+
+            // Assert
+            Assert.Equal(expectedUri, query.ToString());
+            Assert.Equal(8, employee.EmpNumber);
+            Assert.Equal(EmployeeType.PartTime, employee.EmpType);
+        }
+
+        [Fact]
+        public void UseByKeyToFilterByCompositeKeys_WithEnumAsKey_DoNotThrowException()
+        {
+            // Arrange
+            string expectedUri = $"{ServiceRoot}/Employees(EmpNumber=8,EmpType=Microsoft.OData.Client.Tests.Tracking.EmployeeType'PartTime',OrgId=1)";
+
+            string response = @"{
+    ""@odata.context"": ""http://localhost:8007/$metadata#Employees"",
+    ""value"": [
+        {
+            ""EmpNumber"": 8,
+            ""EmpType"": ""PartTime"",
+            ""OrgId"": 1,
+            ""Name"": ""Employee 24""
+        }
+    ]
+}";
+            SetupContextWithRequestPipeline(new DataServiceContext[] { _defaultContext }, response, "Employees");
+
+            // Act
+            EmployeeSingle query = _defaultContext.Employees.ByKey(

```

Comment:
nit: since this is a custom extension method, would be good to assert that the generated request URL is as expected.

Reply:
@habbes I have added the assert for query URL.

[Assert.Equal(expectedUri, query.Query.ToString());](https://github.com/OData/odata.net/pull/3013/files/04d79b127d87e6222d99cec5f14c6ae27cc39b78#diff-925c126a9411dd12bce58e13d4c4c48f79e51bcb1226daa61dae647fcad0263a:~:text=Assert.Equal(expectedUri%2C%20query.Query.ToString())%3B)